packages/server/lib/video_capture.ts

```diff
@@ -274,32 +275,39 @@ export function start (name, options: StartOptions = {}) {
 type OnProgress = (p: number) => void
 
 export async function process (name, cname, videoCompression, ffmpegchaptersConfig, onProgress: OnProgress = function () {}) {
-  const metaFileName = `${name}.meta`
-
-  const maybeGenerateMetaFile = Bluebird.method(() => {
-    if (!ffmpegchaptersConfig) {
-      return false
-    }
-
-    // Writing the metadata to filesystem is necessary because fluent-ffmpeg is just a wrapper of ffmpeg command.
-    return fs.writeFile(metaFileName, ffmpegchaptersConfig).then(() => true)
-  })
-
-  const addChaptersMeta = await maybeGenerateMetaFile()
-
   let total = null
 
+  const metaFileName = `${name}.meta`
+  const addChaptersMeta = ffmpegchaptersConfig && await fs.writeFile(metaFileName, ffmpegchaptersConfig).then(() => true)
+
   return new Bluebird((resolve, reject) => {
     debug('processing video from %s to %s video compression %o',
       name, cname, videoCompression)
 
     const command = ffmpeg()
+    .addOptions([
```

Comment:
This could use some more documentation. It may be a lot of cover in a code comment, but it would be nice to not have an engineer re-learn every single flag a year from now when they look at this code. What would you suggest in terms of documentation for this?

Reply:
I've added a bunch of comments to all these, will push the changes in just a sec.

Reply:
😍 😍 😍